cornice/ext/swagger.py

```diff
@@ -0,0 +1,207 @@
+"""Cornice Swagger 2.0 documentor"""
+
+import re
+import cornice.ext.swagger_model
+import cornice.service
+import cornice.util
+
+
+def schema_to_parameters(schema, service=None):
+    """ Convert Colander Schema object to a Swagger Parameter dict.
+    Provide optional Cornice Service to detect if parameter is in Path
+    """
+    # We want to get all the attributes which aren't specifically
+    # for the body, because swagger deserves them a special
+    # treatment.
+    ret = []
+
+    for location in ("path", "header", "querystring"):
+        attributes = schema.get_attributes(location=location)
+        if attributes and location != "body":
+            for attr in attributes:
+                parameter = dict()
+                name = attr.name
+                paramType = location
+                if paramType == "querystring":
+                    if service and "{%s}" % name in service.path:
+                        paramType = "path"
+                    else:
+                        paramType = "query"
+                parameter["in"] = paramType
+                parameter["name"] = name
+                if hasattr(attr, "description"):
+                    parameter["description"] = getattr(attr, "description")
+                parameter["required"] = getattr(attr, "required")
```

Comment:
This line should be replaced with the following two due to how Colander calculates  "required" with respect to `deferred` Schema Binding.  

        missing = getattr(attr, 'missing')
        parameter['required'] = False if (missing == drop or isinstance(missing, deferred)) else True

[Per the source](https://github.com/Pylons/colander/blob/master/colander/__init__.py#L1983), Colander treats `deferred`  attributes as `required`.  I think the logic should be that if a user defers determining what a `missing` SchemaNode/Parameter value should be, then it is clearly not required as some logic is in place to handle it being missing.